test/api_callable.js

```diff
@@ -523,6 +523,29 @@ describe('bundleable', function() {
         .then(bundledCallback)
         .catch(done);
   });
+
+  it('cancels partially on bundling method', function(done) {
+    var apiCall = createApiCall(func, settings);
+    var expectedSuccess = false;
+    var expectedFailure = false;
+    apiCall(createRequest([1, 2, 3], 'id'), null).then(function(obj) {
+      expect(obj.field1).to.deep.equal([1, 2, 3]);
+      expectedSuccess = true;
+      if (expectedSuccess && expectedFailure) {
+        done();
+      }
+    }).catch(done);
+    var p = apiCall(createRequest([1, 2, 3], 'id'), null);
+    p.then(function(obj) {
+      done(new Error('should not succeed'));
+    }).catch(function(err) {
+      expectedFailure = true;
+      if (expectedSuccess && expectedFailure) {
+        done();
+      }
+    });
```

Comment:
NodeJS runtime is single-threaded, it won't switch to another scheduled task in the middle of the code. Because `apiCall` does not run the task immediately but schedules the task, `p.then` and `p.cancel` have to be executed before the task is executed, even though they took how much time, and therefore, it always needs to be canceled.